frontend/public/components/monitoring/alerting.tsx

```diff
@@ -162,24 +162,54 @@ const alertStateIcons = {
 const AlertStateIcon: React.FC<{ state: string }> = ({ state }) => alertStateIcons[state];
 
 export const AlertState: React.FC<AlertStateProps> = ({ state }) => {
+  const { t } = useTranslation();
+
   const icon = alertStateIcons[state];
+
+  const getStateKey = (stateData) => {
+    switch (stateData) {
+      case AlertStates.Firing:
+        return t('public~Firing');
+      case AlertStates.Pending:
+        return t('public~Pending');
+      case AlertStates.Silenced:
+        return t('public~Silenced');
+      default:
+        return t('public~Not Firing');
+    }
+  };
```

Comment:
I think it would make sense to move `getStateKey` outside of the component (and use `i18next.t`) since it will be used in at least other place.